pkg/controller/cache/managed.go

```diff
@@ -121,134 +123,103 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (resource.E
 	if err != nil {
 		return resource.ExternalObservation{}, errors.Wrap(err, errDescribeReplicationGroup)
 	}
+	rg := rsp.ReplicationGroups[0]
+	elasticache.LateInitialize(&cr.Spec.ForProvider, rg)
+
+	o := resource.ExternalObservation{
+		ResourceExists:    true,
+		ConnectionDetails: resource.ConnectionDetails{},
+	}
 
 	// DescribeReplicationGroups can return one or many replication groups. We
 	// ask for one group by name, so we should get either a single element list
 	// or an error.
-	existing := rsp.ReplicationGroups[0]
-	g.Status.State = aws.StringValue(existing.Status)
-	g.Status.Endpoint = elasticache.ConnectionEndpoint(existing).Address
-	g.Status.Port = elasticache.ConnectionEndpoint(existing).Port
-	g.Status.ProviderID = aws.StringValue(existing.ReplicationGroupId)
-	g.Status.ClusterEnabled = aws.BoolValue(existing.ClusterEnabled)
-	g.Status.MemberClusters = existing.MemberClusters
-
-	switch g.Status.State {
-	case v1alpha2.StatusAvailable:
-		g.Status.SetConditions(runtimev1alpha1.Available())
-		resource.SetBindable(g)
-	case v1alpha2.StatusCreating:
-		g.Status.SetConditions(runtimev1alpha1.Creating())
-	case v1alpha2.StatusDeleting:
-		g.Status.SetConditions(runtimev1alpha1.Deleting())
+	ccList := make([]elasticache2.CacheCluster, len(cr.Status.AtProvider.MemberClusters))
+	for i, cc := range cr.Status.AtProvider.MemberClusters {
```

Comment:
Should we update our observation of the replication group before iterating over its member clusters? I don't recall whether it's possible that we might have more or less clusters member clusters than last time we observed the replication group.